Makefile

```diff
@@ -42,6 +42,16 @@ INSTALL      := install
 prefix       := /usr/local
 bindir       := $(prefix)/bin
 
+help: ## Print help for targets with comments.
+	@echo "Usage:"
+	@echo "  make [target] [FLAG=foo FLAG2=bar...]"
+	@echo ""
+	@echo "Available commands:"
+	@grep -Eh '^[a-zA-Z._-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "  %-30s %s\n", $$1, $$2}'
+	@echo ""
+	@echo "Available flags:"
```

Comment:
Annnnnnd I just caused a conflict. Sorry! Since you'll have to rebase anyway, two optional nits:

1. These might be better described as "Useful {commands|flags}", since they're not exhaustive.
2. I'd write `make [target...] [VAR=foo VAR2=bar]` and "useful variables" since they're not technically flags, but perhaps I'm too deep in Make-land to know what would be more useful to users.